manifests/charts/ai-gateway-helm/values.yaml

```diff
@@ -67,6 +67,11 @@ controller:
     # If not set and create is true, a name is generated using the fullname template
     name: ""
 
+  # -- ClusterRole --
+  # Whether to add a suffix to clusterRole and clusterRoleBinding.
+  # Note: this is mostly for the backward compatibility, and defaults to false.
+  suffixClusterRole: false
```

Comment:
I think we can do better for users to keep backward compatibility:

* We could check if the cluster named after a service account exists, and if not, create the new ClusterRole with a consistent, unique name.
* If the cluster named after the service account exists, use it.

This way we transparently handle backwards compatibility in a graceful way without delegating that complexity to users.

Reply:
+1

Reply:
FWIW, [here you have an example](https://github.com/envoyproxy/ai-gateway/blob/908fae31d463078b24700a4e124cd903f0aba789/manifests/charts/ai-gateway-helm/templates/admission_webhook.yaml#L63) of how to check if a resource exists in Helm.

Reply:
Yes, this method is better.